src/nuxt/sdk-setup.ts

```diff
@@ -0,0 +1,209 @@
+// @ts-expect-error - clack is ESM and TS complains about that. It works though
+import clack from '@clack/prompts';
+import * as Sentry from '@sentry/node';
+import chalk from 'chalk';
+import fs from 'fs';
+// @ts-expect-error - magicast is ESM and TS complains about that. It works though
+import { loadFile, generateCode, MagicastError } from 'magicast';
+// @ts-expect-error - magicast is ESM and TS complains about that. It works though
+import { addNuxtModule } from 'magicast/helpers';
+import path from 'path';
+import {
+  getConfigBody,
+  getDefaultNuxtConfig,
+  getNuxtModuleFallbackTemplate,
+  getSentryConfigContents,
+} from './templates';
+import {
+  abort,
+  abortIfCancelled,
+  featureSelectionPrompt,
+  isUsingTypeScript,
+} from '../utils/clack-utils';
+import { traceStep } from '../telemetry';
+
+const possibleNuxtConfig = [
+  'nuxt.config.js',
+  'nuxt.config.mjs',
+  'nuxt.config.cjs',
+  'nuxt.config.ts',
+  'nuxt.config.mts',
+  'nuxt.config.cts',
+];
+
+export async function getNuxtConfig(): Promise<string> {
+  let configFile = possibleNuxtConfig.find((fileName) =>
+    fs.existsSync(path.join(process.cwd(), fileName)),
+  );
+
+  if (!configFile) {
+    clack.log.info('No Nuxt config file found, creating a new one.');
+    Sentry.setTag('nuxt-config-strategy', 'create');
+    // nuxt recommends its config to be .ts by default
+    configFile = 'nuxt.config.ts';
+
+    await fs.promises.writeFile(
+      path.join(process.cwd(), configFile),
+      getDefaultNuxtConfig(),
+      { encoding: 'utf-8', flag: 'w' },
+    );
+
+    clack.log.success(`Created ${chalk.cyan('nuxt.config.ts')}.`);
+  }
+
+  return path.join(process.cwd(), configFile);
+}
+
+export async function addSDKModule(
+  config: string,
+  options: { org: string; project: string; url: string; selfHosted: boolean },
+): Promise<void> {
+  try {
+    const mod = await loadFile(config);
+
+    addNuxtModule(mod, '@sentry/nuxt/module', 'sentry', {
+      sourceMapsUploadOptions: {
+        org: options.org,
+        project: options.project,
+        ...(options.selfHosted && { url: options.url }),
+      },
+    });
+    addNuxtModule(mod, '@sentry/nuxt/module', 'sourcemap', {
+      client: 'hidden',
+    });
+
+    const { code } = generateCode(mod);
+
+    await fs.promises.writeFile(config, code, { encoding: 'utf-8', flag: 'w' });
+
+    clack.log.success(
+      `Added Sentry Nuxt Module to ${chalk.cyan(path.basename(config))}.`,
+    );
+  } catch (e: unknown) {
+    // Cases where users spread options are not covered by magicast,
+    // so we fall back to showing how to configure the nuxt config
+    // manually.
+    if (e instanceof MagicastError) {
+      clack.log.warn(
+        `Automatic configuration of ${chalk.cyan(
+          path.basename(config),
+        )} failed, please add the following settings:`,
+      );
+      // eslint-disable-next-line no-console
+      console.log(`\n\n${getNuxtModuleFallbackTemplate(options)}\n\n`);
```

Comment:
l (feel free to disregard if N/A): We have a diff-like helper function for displaying code snippets: `makeCodeSnippet` in `clack-utils.ts`. Might be useful here but I'll let you make the call if it's necessary.

Reply:
Thanks, I'll disregard for now to get this out, but will revisit.